lib/workers/pr/changelog/github/index.ts

```diff
@@ -52,18 +54,30 @@ export async function getReleaseNotesMd(
 
   // https://docs.github.com/en/rest/reference/git#get-a-tree
   const res = await http.getJson<GithubGitTree>(
-    `${apiPrefix}/git/trees/${defaultBranch}`
+    `${apiPrefix}/git/trees/${defaultBranch}${
+      sourceDirectory ? '?recursive=1' : ''
```

Comment:
If there are too many. Files in repower don't find what we like. Isn't it better to search in root first and fallback to source directory? So we would have one additional query?

Maybe we can cache the found file path per dep?

Reply:
We only search recursively if there's been a sourceDirectory configured, which is not that common and unlikely to be incorrect (because not even most monorepo authors know to use it). And the recursive query is actually a single API call - it's not like GitHub's usual. I think this is reasonably efficient for now

Reply:
If/when we support fetching and parsing arbitrary changelogUrls (today we just link to them) then we could also consider improvements here too.